geode-core/src/main/java/org/apache/geode/distributed/internal/direct/DirectChannel.java

```diff
@@ -112,6 +115,8 @@ public DirectChannel(Membership<InternalDistributedMember> mgr,
       throws ConnectionException {
     this.receiver = listener;
     this.dm = dm;
+    this.stats = dm.getStats();
+    this.bufferPool = new BufferPool(stats);
```

Comment:
now `DirectChannel` owns `BufferPool` and makes the pool accessible as needed...